### PR TITLE
Cleanup psd example.

### PR DESCRIPTION
**Pull request by anntzer**
- Show all figures at once.
- Group axes setter calls to have them take less room relative to the
  psd calls, which are the object of the example.
- Align the noverlap examples to make the parallelism between them
  clearer.
- Don't introduce a second random state with a single use.

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: examples/lines_bars_and_markers/psd_demo.py [hunk 1]
```diff
@@ -57,43 +55,35 @@
 y = 10. * np.sin(2 * np.pi * 4 * t) + 5. * np.sin(2 * np.pi * 4.25 * t)
 y = y + np.random.randn(*t.shape)
 
-# Plot the raw time series
+# Plot the raw time series.
 fig = plt.figure(constrained_layout=True)
 gs = gridspec.GridSpec(2, 3, figure=fig)
```
[the remark this answers is not in the document]
Comment (anntzer) [reply examples/lines_bars_and_markers/psd_demo.py]: I switched to the more idiomatic add_gridspec, but switching to subplot_mosaic would both result in slightly inconsistent layouts in this subexample relative to all others (unless you make the change in all subexamples) and 2) make the lines below no longer fit in 79 characters.  I wouldn't be opposed if you make such a global change across all subexamples of the file, but let's not do this here?

**Comment by jklymak** [comment]
We are trying to deprecate these, so why are we fixing the example?

**Comment by anntzer** [comment]
We've been talking about deprecating this for a long time and I don't think that'll happen any time soon (though I'd be happy to be proven wrong) so I think in the meantime fix improvements are still useful (unless we decide to "deprecate by removing from the docs", which may be a valid strategy too...).

**Comment by timhoffm** [comment]
> This seems good, but I thought we had a discussion about using `set` over `set_*`, and I don't know where we landed. Maybe @timhoffm remembers?
 
The status is (https://github.com/matplotlib/matplotlib/pull/20631#issuecomment-881007639):
- `set()` and `set_*()` will both be supported, i.e. no deprecations
- we need better documentation on the concepts - I just read in the above linke I volunteered volunteered to write this 👀 
- we have not decided, whether/where we want to use `set()` in the docs.

Because of the last point, I would currently not activately switch to `set()`. Nevertheless, In my eyes it's helpful to switch to `set()` in many examples where all the `set_*` are not a central aspect but only added to make the plot look better. - It's less distracting if they are all tucked away in a single call.

**Comment by jklymak** [comment]
> We've been talking about deprecating this for a long time and I don't think that'll happen any time soon (though I'd be happy to be proven wrong) so I think in the meantime fix improvements are still useful (unless we decide to "deprecate by removing from the docs", which may be a valid strategy too...). 

There is a PR to deprecate it...

**Comment by anntzer** [comment]
Ah, sorry I missed that.  Let's just not bother, then.

**Pull request closed without merging**